### DISCUSSION TITLE
How can you get the execution logs of a job?

### DISCUSSION THREAD

**Discussion by totycro** [EOEPCA/eoepca#80] [ADES (Processing)]
Sometimes jobs fail and it's necessary to have a look at the logs. Since the job namespace is deleted after the job run, it's not possible to do that directly via `kubectl logs`. How can you access the job logs via ades?

**Comment by rconway** [comment]
@bbrauzzi Can you advise

**Reply by bbrauzzi** [reply]
Hi @totycro 
the logs of the completed/failed processing jobs are available from the URI path _/\_run/res_.
( http://HOSTNAME:PORT/WORKSPACE_NAME/_run/res )
The application log files are the ones ending with __calrissian.log_
Also the _workflow_executor.log_ can help the debugging when there is a kubernetes resource issue.
We are currently working to improve and ease the user logs access from the Ades interface.
Kind regards
Blasco

**Reply by totycro** [reply]
I've now tried this URL, but it gives a 404 Not Found:
`http://localhost:8080/bmtest/_run/res`

However I do get a response at this endpoint:
`http://localhost:8080/bmtest/wps3/jobs/a9831344-bcc5-11ed-8929-a2df302099b9/_run/res`

It's the same response that I get under `http://localhost:8080/bmtest/wps3/jobs/a9831344-bcc5-11ed-8929-a2df302099b9/results`, which is:
```json
{
  "s3_catalog_output": "s3://bm-test-udoo2oh/wf-a9831344-bcc5-11ed-8929-a2df302099b9/catalog.json"
}
```

This URL also returns the same job result data:
`http://localhost:8080/bmtest/wps3/jobs/a9831344-bcc5-11ed-8929-a2df302099b9/_calrissian.log`


How can i find the logs from this?

**Reply by bbrauzzi** [reply]
Sorry, my fault, the path to the logs folder should not include the WORKSPACE_NAME
Try the URL _http://localhost:8080/_run/res_ , this URL will return the list of all the job logs.

In this list you should find the log of your job whose filename should end in
_a9831344-bcc5-11ed-8929-a2df302099b9_0_calrissian.log_

**Reply by totycro** [reply]
Thanks, I now found the logs here:
`http://localhost:8080/_run/res/snuggs-0-3-0a9831344-bcc5-11ed-8929-a2df302099b9_0_calrissian.log`

There are also some more files with job statistics and other information.
